### PR TITLE
Rename `namespace_registry` to `registry`

### LINKED ISSUE #477: Rename `namespace-registry` to `registry` to be distinguished from `namespace`
Right now it's very confusing to people between `namespace-registry` and `namespace` 
So @jhiemstrawisc and I discussed that we should at some point put some effort to make it clear. So I propose we rename `namespace-registry` to `registry` to emphasize that registry is where to register the namespaces. Or it's very welcomed if there's some other better approaches to this issue.

### PR DESCRIPTION
**Pull request by haoming29**
Closes #477

### CODE REVIEW

File: namespaces/namespaces.go [hunk 1]
```diff
@@ -16,6 +16,13 @@
  *
  ***************************************************************/
 
+// Package namespaces implements namespace lookups and matches for legacy [stashcp] and [osdf-client]
+// to maintain backward compatibility.
+//
+// The namespaces pacakge should not be used for any new features Pelican introduces.
+//
```
Comment (joereuss12) [inline namespaces/namespaces.go:23]: Just a small typo here for "package"

File: registry/registry.go [hunk 2]
```diff
@@ -16,7 +16,15 @@
...
 
-package nsregistry
+// Package registry handles namespace management in Pelican ecosystem.
+//
+//   - It handles the logic to spin up a "registry" server for namespace management,
+//     including a web UI for interactive namespace registration, approval, and browsing.
+//   - It provides a CLI tool `./pelican namespace <command> <args>` to list, register, and delete a namespace
+//
+// To register a namespace, first spin up registry server by `./pelican registry -p <your-port-number>`, and then use either
+// the CLI tool or go to registry web UI at `https://localhost:<your-port-number>/view/`, and follow instructions for next steps.
```
Comment (joereuss12) [inline registry/registry.go:26]: Isn't it "./pelican registry **_serve_** -p ..."?

**Review by joereuss12 — CHANGES_REQUESTED** [review]
LGTM, just address those small typos

**Comment by haoming29** [comment]
> LGTM, just address those small typos

Thanks for catching that!

**Review by joereuss12 — APPROVED** [review]
LGTM

**Pull request merged**